File: java/src/main/java/com/hydrapayments/sdk/model/CreateIntentResponse.java
package com.hydrapayments.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CreateIntentResponse {

    private String id;
    private String status;
    private int amount;
    private String currency;
    private String clientSecret;

    public CreateIntentResponse() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public int getAmount() { return amount; }
    public void setAmount(int amount) { this.amount = amount; }

    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }

    @JsonProperty("client_secret")
    public String getClientSecret() { return clientSecret; }
    @JsonProperty("client_secret")
    public void setClientSecret(String clientSecret) { this.clientSecret = clientSecret; }
}

File: dotnet/src/HydraPayments.Sdk/Models/CreateIntentResponse.cs
using System.Text.Json.Serialization;

namespace HydraPayments.Sdk.Models;

public class CreateIntentResponse
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("status")]
    public string Status { get; set; } = string.Empty;

    [JsonPropertyName("amount")]
    public int Amount { get; set; }

    [JsonPropertyName("currency")]
    public string Currency { get; set; } = string.Empty;

    [JsonPropertyName("client_secret")]
    public string ClientSecret { get; set; } = string.Empty;
}
